Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1431,63 +1444,22 @@ namespace xr
                 }
 
                 // Get gamepad data 
-                const auto& controllerInfo = sessionImpl.ControllerInfo;
-                if (controllerInfo.ControllerBinding)
                 {
-                     auto& inputSource = InputSources[idx];
-                     inputSource.GamepadTrackedThisFrame = true;
-
-                    // Get trigger value data
-                    float currentTriggerValue;
-                    if (m_impl->QueryControllerFloatAction(actionResources.ControllerGetTriggerValueAction, session, currentTriggerValue))
-                    {
-                        inputSource.GamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value = currentTriggerValue;
-                    }
-
-                    // Get squeeze click data
-                    bool currentSqueezeValue;
-                    if (m_impl->QueryControllerBooleanAction(actionResources.ControllerGetSqueezeClickAction, session, currentSqueezeValue))
-                    {
-                        inputSource.GamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed = currentSqueezeValue;
-                    }
-
-                    // Get trackpad axes data
-                    float currentTrackpadXAxis, currentTrackpadYAxis;
-                    if (m_impl->QueryControllerVector2fAction(actionResources.ControllerGetTrackpadAxesAction, session, currentTrackpadXAxis, currentTrackpadYAxis))
+                    const auto& controllerInfo = sessionImpl.ControllerInfo;
+                    auto gamepadObject = InputSources[idx].GamepadObject;
+
+                    // Update gamepad data
+                    if ((m_impl->TryUpdateControllerFloatAction(actionResources.ControllerGetTriggerValueAction, session, gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value)) &&
```

Comment:
>TryUpdateControllerFloatAction [](start = 33, length = 30)

nit: can any of these fail while others succeed?

Reply:
I don't think so, but if any do fail I'd rather not send incomplete data (since babylonJS consumes all or none of it)